src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -522,18 +558,32 @@ export default {
 			this.$refs.openConversationsList.showModal()
 		},
 
+		isPreviousFilter() {
+			const cachedPreviousFilter = this.previousFilter
+			this.previousFilter = this.isFiltered
+			return cachedPreviousFilter
+		},
+
 		handleFilter(filter) {
 			this.isFiltered = filter
-
 			// Store the active filter
 			if (filter) {
 				BrowserStorage.setItem('filterEnabled', filter)
 			} else {
 				BrowserStorage.removeItem('filterEnabled')
 			}
-
 			// Clear the search input once a filter is active
 			this.searchText = ''
+			// Initiate the previous filter
+			this.previousFilter = null
+		},
+
+		filterFunction(conversation) {
```

Comment:
Maybe, move this function to util completely, to have only one entrypoint from another file?